Api/CalcWrapper.cs

```diff
@@ -0,0 +1,298 @@
+            loadProperties[1].Name = "ReadOnly";
+            loadProperties[1].Value = new uno.Any(false);
+
+            if (Hidden)
+            {
+                loadProperties[0].Name = "Hidden";
+                loadProperties[0].Value = new uno.Any(true);
+
+                //Check to see if the path is provided by the user, if it's empty, create a new document. 
+                if (String.IsNullOrEmpty(Path))
+                {
+                    xComponent = aLoader.loadComponentFromURL("private:factory/scalc", "_blank", 0, loadProperties);
+                }
+                else
+                {
+                    try
+                    {
+                        Path = String.Format("file:///{0}", Path);
+                        xComponent = aLoader.loadComponentFromURL(Path, "_blank", 0, loadProperties);
+                    }
+                    catch (unoidl.com.sun.star.uno.Exception ex)
+                    {
+                        throw new unoidl.com.sun.star.uno.Exception(ex.Message, ex);
+                    }
+                }
+            }
+            else
+            {
+                loadProperties[0].Name = "Hidden";
+                loadProperties[0].Value = new uno.Any(false);
+                if (String.IsNullOrEmpty(Path))
+                {
+                    xComponent = aLoader.loadComponentFromURL("private:factory/scalc", "_blank", 0, new unoidl.com.sun.star.beans.PropertyValue[0]);
+                }
+                else
+                {
+                    try
+                    {
+                        Path = String.Format("file:///{0}", Path);
+                        xComponent = aLoader.loadComponentFromURL(Path, "_blank", 0, new unoidl.com.sun.star.beans.PropertyValue[0]);
+                    }
+                    catch (unoidl.com.sun.star.uno.Exception ex)
+                    {
+                        throw new unoidl.com.sun.star.uno.Exception(ex.Message, ex);
+                    }
+                }
+            }
+            return (unoidl.com.sun.star.sheet.XSpreadsheetDocument)xComponent;
+        }
+
+        private void SaveDocument(String Path)
+        {
+            try
+            {
+                Path = Path.Replace("\\", "/"); // Convert forward slashes to backslashes, converting it to the correct format storeToURL expects. 
+                Path = String.Concat("file:///", Path);
+                XStorable xStorable = (XStorable)mxDocument; // Typecast the currently open document to XStorable type.
+                xStorable.storeToURL(Path, new unoidl.com.sun.star.beans.PropertyValue[1]); //Creating an empty PropertyValue array saves the document in the default .ods format.
+            }
+            catch(unoidl.com.sun.star.uno.Exception ex)
+            {
+                throw new unoidl.com.sun.star.uno.Exception(ex.Message, ex);
+            }
+        }
+
+        private void AddNewSheet(String Name)
```

Comment:
string name; same issue with most of other arguments, please correct it also elsewhere.